Source/ACE/Entity/Player.cs

```diff
@@ -536,7 +536,8 @@ public QueuedGameAction ExaminationQueuePop()
         public void GrantXp(ulong amount)
         {
             // until we are max level we must make sure that we send
-            var chart = DatabaseManager.Charts.GetLevelingXpChart();
+            XpTable xpTable = XpTable.ReadFromDat();

```

Comment:
These should be read once and cached in a static variable.  I'd rather have higher memory requirements than create performance issues of reading from dat files every time somebody clicks the up arrow.